lib/polyfill/patchedmediakeys_20140218.js

```diff
@@ -0,0 +1,800 @@
+ * 12 March 2015} on browsers with EME version
+ * http://www.w3.org/TR/2014/WD-encrypted-media-20140218/.
+ *
+ * @see http://enwp.org/polyfill
+ */
+
+goog.provide('shaka.polyfill.PatchedMediaKeys.v20140218');
+goog.require('shaka.asserts');
+goog.require('shaka.log');
+goog.require('shaka.util.EventManager');
+goog.require('shaka.util.FakeEvent');
+goog.require('shaka.util.FakeEventTarget');
+goog.require('shaka.util.Pssh');
+goog.require('shaka.util.PublicPromise');
+
+
+/**
+ * Install the polyfill.
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.install = function() {
+  shaka.log.debug('PatchedMediaKeys.v20140218.install');
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  // Delete mediaKeys to work around strict mode compatibility issues.
+  delete HTMLMediaElement.prototype['mediaKeys'];
+  // Work around read-only declaration for mediaKeys by using a string.
+  HTMLMediaElement.prototype['mediaKeys'] = null;
+  HTMLMediaElement.prototype.setMediaKeys = v20140218.setMediaKeys;
+
+  // Install patches
+  window.MediaKeySession = v20140218.MediaKeySession;
+  window.MediaKeys = v20140218.MediaKeys;
+  window.MediaKeySystemAccess = v20140218.MediaKeySystemAccess;
+  Navigator.prototype.requestMediaKeySystemAccess =
+      v20140218.requestMediaKeySystemAccess;
+};
+
+
+/**
+ * An implementation of Navigator.prototype.requestMediaKeySystemAccess.
+ * Retrieve a MediaKeySystemAccess object.
+ *
+ * @this {!Navigator}
+ * @param {string} keySystem
+ * @param {!Array.<!MediaKeySystemConfiguration>} supportedConfigurations
+ * @return {!Promise.<!MediaKeySystemAccess>}
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.requestMediaKeySystemAccess =
+    function(keySystem, supportedConfigurations) {
+  shaka.log.debug('PatchedMediaKeys.v20140218.requestMediaKeySystemAccess');
+  shaka.asserts.assert(this instanceof Navigator);
+
+  // Alias.
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+  try {
+    var access = new v20140218.MediaKeySystemAccess(
+        keySystem, supportedConfigurations);
+    return Promise.resolve(/** @type {!MediaKeySystemAccess} */ (access));
+  } catch (exception) {
+    return Promise.reject(exception);
+  }
+};
+
+
+
+/**
+ * An implementation of MediaKeySystemAccess.
+ *
+ * @constructor
+ * @param {string} keySystem
+ * @param {!Array.<!MediaKeySystemConfiguration>} supportedConfigurations
+ * @implements {MediaKeySystemAccess}
+ * @throws {Error} if the key system is not supported.
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySystemAccess =
+    function(keySystem, supportedConfigurations) {
+  shaka.log.debug('v20140218.MediaKeySystemAccess');
+
+  /** @type {string} */
+  this.keySystem = keySystem;
+
+  /** @private {!MediaKeySystemConfiguration} */
+  this.configuration_;
+
+  // This is only a guess, since we don't really know from the prefixed API.
+  var allowPersistentState = true;
```

Comment:
`persistentState` is required to create `persistent-license` sessions, but the opposite does not apply.

If you aren't sure that a DRM implementation does not persist state, the browser should disallow  `persistentState=="not-allowed"`. (The same is true for `distinctiveIdentifier`.)

It looks like this is just used to report the assumed result. In that case, true is correct. It would probably help to rename the variable to indicate it is a property of the key system. You probably also want to reject a configuration with `not-allowed"`.

You also need to do the same thing for `distinctiveIdentifier`.